node/runtime/src/lib.rs

```diff
@@ -906,8 +906,25 @@ impl Runtime {
                         );
                         Ok(vec![])
                     },
-                    ReceiptBody::ManaAccounting(_mana_accounting) => {
-                        // TODO(#259): Refund mana and charge gas
+                    ReceiptBody::ManaAccounting(mana_accounting) => {
+                        let key = get_tx_stake_key(
+                            &mana_accounting.accounting_info.originator,
+                            &mana_accounting.accounting_info.contract_id,
+                        );
+                        let tx_total_stake: Option<TxTotalStake> = get(state_update, &key);
+                        if let Some(mut tx_total_stake) = tx_total_stake {
+                            let config = TxStakeConfig::default();
+                            tx_total_stake.update(block_index, &config);
+                            tx_total_stake.refund_mana_and_charge_gas(
+                                mana_accounting.mana_refund,
+                                mana_accounting.gas_used,
+                                &config,
+                            );
+                            set(state_update, &key, &tx_total_stake);
+                        } else {
+                            // TODO(#445): Figure out what to do when the TxStake doesn't exist during mana accounting
+                            panic!("TX stake doesn't exist when mana accounting arrived");
```

Comment:
It's better to use `unimplemented` here

Reply:
It actually panic. Since we don't know how to handle it. I think we're going to change economics to pay upfront for TX to avoid this problem